packages/shared-business/src/components/UploadArea.tsx

```diff
@@ -114,13 +115,27 @@ const styles = StyleSheet.create({
   },
 });
 
-export type UploadFileStatus = {
-  id: string;
-  name?: string;
-  fileUrl?: string;
-  progress?: number;
-  finished?: boolean;
-};
+export type UploadFileStatus =
```

Comment:
a trick: it could be

```ts
export type UploadFileStatus =
  | {
      id: string;
      name?: string;
      fileUrl?: string;
    } & (
      | { status: "UPLOADING"; progress: number }
      | { status: "FINISHED" }
      | { status: "FAILED"; error: string }
    );
```

Reply:
done, thanks